### ISSUE TITLE
tax contribution calculation

### ISSUE THREAD

**Issue by Dharr-Guartonzec** [evecm/ecm#21]
Maybe another issue here or i just mess up with understanding the mechanics.

![tax-contribution-may-2015](https://cloud.githubusercontent.com/assets/11791747/7920059/bba41afe-089c-11e5-8cd0-b9802199185b.png)

Check that uploaded screenshot from the tax contribution. I selected the month may and the amount of tax you see on top in bold does not match the total when you just roughly add all the displayed member contributions together.

Why is that? Or is the top total maybe for the selected peiod of month may, while the rows with the single member amounts show something different? Like alltime, which is the default filter when you enter the tax contribution page... is just a guess - i have no clue.

**Comment by ggrog** [comment]
Yes, that doesn't look right.  A cursory look didn't reveal anything to me and I don't have a non-zero tax corp to test with.  Looks like the rest of the numbers add up, just not the total line at the top.

Any clues when you play with the dates?  Does the total change at all?  What if you make the date range just one or two days?  Can you find a date ranges where it matches but not others?  If you find a specific date, perhaps there's an entry in the full journal that is obviously skewing the total?  A wallet withdrawl of some sort maybe?

A possible theory based on looking at the code - it looks like the total is inclusive of the dates in the picker, but the lists are generated using dates exclusively between the range given.  However, this would cause the total to be higher than the individual list.

**Comment by Dharr-Guartonzec** [comment]
Her is it for just 2 days, and the numbers below the total at the top are more ISK then top shows for this filtered period.

![tax-2days-only](https://cloud.githubusercontent.com/assets/11791747/7921251/89a2f1bc-08a4-11e5-9d58-08678cfc3aff.png)

**Comment by ggrog** [comment]
Check the journals for something during those two days that might explain the difference.

**Comment by Dharr-Guartonzec** [comment]
oha, you mean total tax income is not only calculated by bouty prizes? I will have a look.

**Comment by ggrog** [comment]
Yes - these are the items that are included:  (The #s are the internal codes)

    16, # Bounty
    17, # Bounty Prize
    33, # Agent Mission Reward
    34, # Agent Mission Time Bonus Reward
    85, # Bounty Prizes
    99, # Corporate Reward Payout
    103, # Corporate Reward Tax

**Comment by Dharr-Guartonzec** [comment]
aha - can it be a TZ issue? I just selected to show me only 1st of june and i also get shown journal entries from yesterday 10:00pm. So maybe some transactions from day before get calculated in...

**Comment by ggrog** [comment]
Ah hah, I think you've nailed it.  Try setting your server TZ to UTC to match EVE.
http://www.thegeekstuff.com/2010/09/change-timezone-in-linux/

(There's code to switch to your local TZ when dealing with accounting entries, but they forgot it in a couple places.  Why the code is there in the first place doesn't make sense, EVE players use EVE time!)

**Comment by Dharr-Guartonzec** [comment]
ok, server is running on UTC now - for the journal filter selection that worked - now i see only exact the same day - but tax total is still less then the below listed values... no clue

**Comment by tyler274** [comment]
Don't know if its related but if you press the refresh button without changing the date (or change the date and press refresh again) the value will change.

**Comment by m0gg** [comment]
Somehow this does not look correct or am i wrong?

![tax_report_mismatch](https://cloud.githubusercontent.com/assets/4560028/9663545/453bf6da-5265-11e5-86b1-7a6d2d184ec0.png)

**Comment by ggrog** [comment]
m0gg I believe you're running into the same issue - there's a bug with timezones and date ranges.
